packages/@aws-cdk/integ-runner/README.md

```diff
@@ -0,0 +1,276 @@
+  readonly preDeploy?: string[];
+  readonly postDeploy?: string[];
+  readonly preDestroy?: string[];
+  readonly postDestroy?: string[];
+}
+
+interface TestCase {
+  /**
+   * Stacks that should be tested as part of this test case
+   * The stackNames will be passed as args to the cdk commands
+   * so dependent stacks will be automatically deployed unless
+   * `exclusively` is passed
+   */
+  readonly stacks: string[];
+
+  /**
+   * Run update workflow on this test case
+   * This should only be set to false to test scenarios
+   * that are not possible to test as part of the update workflow
+   *
+   * @default true
+   */
+  readonly update?: boolean;
+
+  /**
+   * Additional options to use for each CDK command
+   *
+   * @default - runner default options
+   */
+  readonly cdkCommandOptions?: CdkCommands;
+  
+  /**
+   * Additional commands to run at predefined points in the test workflow
+   *
+   * e.g. { postDeploy: ['yarn', 'test'] }
+   *
+   * @default - no hooks
+   */
+  readonly hooks?: Hooks;
+  
+  /**
+   * Whether or not to include asset hashes in the diff
+   * Asset hashes can introduces a lot of unneccessary noise into tests,
+   * but there are some cases where asset hashes _should_ be included. For example
+   * any tests involving custom resources or bundling
+   * 
+   * @default false
+   */
+  readonly diffAssets?: boolean;
+  
+  /**
+   * List of CloudFormation resource types in this stack that can
+   * be destroyed as part of an update without failing the test.
+   *
+   * This list should only include resources that for this specific
+   * integration test we are sure will not cause errors or an outage if
+   * destroyed. For example, maybe we know that a new resource will be created
+   * first before the old resource is destroyed which prevents any outage.
+   *
+   * e.g. ['AWS::IAM::Role']
+   *
+   * @default - do not allow destruction of any resources on update
+   */
+  readonly allowDestroy?: string[];
+  
+  /**
+   * Limit deployment to these regions
+   *
+   * @default - can run in any region
+   */
+  readonly regions?: string[];
+}
+
+interface IntegSchema {
+  /**
+   * Test cases that the runner will execute
+   */
+  testCases: { [name: string]: TestCase };
+}
+```
+
+### defining an integration test
+
+In most cases an integration test will be an instance of a stack
+
+```ts
+import { Function, FunctionOptions } from '../lib';
+
+interface MyIntegTestProps extends StackOptions {
+  functionProps?: FunctionOptions;
+}
+class MyIntegTest extends Stack {
+  constructor(scope: Construct, id: string, props: MyIntegTestProps) {
+    super(scope, id, props);
+	
+	new Function(this, 'Handler', {
+	  runtime: Runtime.NODEJS_12_X,
+	  handler: 'index.handler',
+	  code: Code.fromAsset(path.join(__dirname, 'lambda-handler')),
+	  ...props.functionProps,
+	});
+  }
+}
+```
+
+You would then use the `IntegTest` construct to create your test cases
+
+```ts
+new IntegTeset(app, 'ArmTest', {
```

Comment:
This could be a `Construct` that uses `attachCustomSynthesis` to synthesize the manifest.